src/services/pool.ts

```diff
@@ -109,9 +143,42 @@ const mapBranches = (branches: BranchEntry[], fullBaseUrl: string) => {
   }));
 };
 
-const mapRepos = (repo: RepoDocument): RepoResponse => ({
-  repoName: repo.repoName,
-  project: repo.project,
-  search: repo.search,
-  branches: mapBranches(repo.branches, getRepoUrl(repo.url[ENV_URL_KEY], repo.prefix[ENV_URL_KEY])),
-});
+const mapRepos = async (repo: RepoDocument) => {
+  try {
+    const query: Filter<DocsetDocument> = { project: repo.project };
+    const strictOptions: FindOptions = {
+      projection: {
+        project: 1,
+        url: 1,
+        prefix: 1,
+      },
+    };
+
+    const repo_docset = await db
+      .collection<DocsetDocument>(DOCSETS_COLLECTION)
```

Comment:
> not sure if this fix is the cleanest way to do this since it involves querying from the database two separate times

I think we can have something similar to the aggregation that the [Autobuilder](https://github.com/mongodb/docs-worker-pool/blob/98a78de87b2a5abb8b5b00eac2e24e639e534a12/src/repositories/docsetsRepository.ts#L13) currently does to help keep logic generally consistent. It might also save a little on the back-and-forth between finds.

That being said, I don't think there's anything wrong with keeping this as multiple finds as long as we don't see any major degradation in performance.